src/nvim/option.c

```diff
@@ -1874,14 +1875,18 @@ static void apply_optionset_autocmd(OptIndex opt_idx, int opt_flags, OptVal oldv
     set_vim_var_string(VV_OPTION_COMMAND, "modeline", -1);
     set_vim_var_tv(VV_OPTION_OLDLOCAL, &oldval_tv);
   }
-  apply_autocmds(EVENT_OPTIONSET, options[opt_idx].fullname, NULL, false, NULL);
+
+  WITH_AUCMD_CONTEXT(ctx.buf, {
+    apply_autocmds(EVENT_OPTIONSET, options[opt_idx].fullname, NULL, false, NULL);
+  });
```

Comment:
is this a behavior change and if so is there a way to cover it with a test?

Reply:
It's not really a behavior change. Previously this context change happened throughout the entirety of `set_option`, now the context change has been narrowed down to the specific parts that need it.